### PR TITLE
Security tests for OffscreenCanvas texImage functions

### PR DESCRIPTION
**Pull request by xlai-o**
@kenrussell  Adding two security tests:
1. sdk/tests/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html tests whether texImage2D and texSubImage2D functions in OffscreenCanvas Webgl context correctly throw security exception when cross-origin image sources are updated
2. sdk/tests/conformance2/textures/misc/origin-clean-conformance-offscreencanvas.html tests whether texImage2D, texImage3D, texSubImage2D, texSubImage3D in OffscreenCanvas Webgl2 context correctly throw security exceptions.

Note that although these two tests look similar to sdk/tests/conformance/textures/misc/origin-clean-conformance.html (for HTMLCanvasElement), they have many differences: (1) toDataURL is not present
in OffscreenCanvas so not tested (2) the functions in webgl2 are completely different from those in webgl1 (3) OffscreenCanvas is used instead of HTMLCanvasElement and test messages are all changed.

### CODE REVIEW

File: sdk/tests/conformance/textures/misc/00_test_list.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ copy-tex-image-and-sub-image-2d.html
 gl-pixelstorei.html
 gl-teximage.html
 origin-clean-conformance.html
+origin-clean-conformance-offscreencanvas.html
```
Comment (kenrussell) [inline sdk/tests/conformance/textures/misc/00_test_list.txt]: Could you please add --min-version 1.0.4 here? This has been done for a while, just to track in which conformance version new tests were added.

File: sdk/tests/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+var defaultImgUrl = "https://get.webgl.org/conformance-resources/opengl_logo.jpg";
+var localImgUrl = "../../../resources/opengl_logo.jpg";
+
+var imgDomain;
+var pageDomain;
+var successfullyParsed;
+
+function endTest() {
```
Comment (kenrussell) [inline sdk/tests/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html]: finishTest() in js-test-pre.js does this; could you please use that instead?

File: sdk/tests/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            "texSubImage2D with NON origin clean canvas should throw exception.");
+
+  debug("check that readPixels continues to work against this offscreencanvas.");
+  assertMsg(!causedException(makeReadPixels(gl)),
+            "readPixels should never throw exception -- not possible to dirty origin of WebGL canvas.");
+
+  // TODO: Should check video.
+  // TODO: Should check CORS support.
+
+  endTest();
```
Comment (kenrussell) [inline sdk/tests/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html]: Could you call finishTest() here instead?

File: sdk/tests/conformance2/textures/misc/00_test_list.txt [hunk 4]
```diff
@@ -12,6 +12,7 @@ gl-get-tex-parameter.html
 mipmap-fbo.html
 --min-version 2.0.1 npot-video-sizing.html
 --min-version 2.0.1 tex-3d-mipmap-levels-intel-bug.html
+origin-clean-conformance-offscreencanvas.html
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/00_test_list.txt]: Could you add --min-version 2.0.1 here?

File: sdk/tests/conformance2/textures/misc/origin-clean-conformance-offscreencanvas.html [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+var defaultImgUrl = "https://get.webgl.org/conformance-resources/opengl_logo.jpg";
+var localImgUrl = "../../../resources/opengl_logo.jpg";
+
+var imgDomain;
+var pageDomain;
+var successfullyParsed;
+
+function endTest() {
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/origin-clean-conformance-offscreencanvas.html]: finishTest() should do all this already.

File: sdk/tests/conformance/textures/misc/00_test_list.txt (hunk 1 above)
Comment (xlai-o) [inline sdk/tests/conformance/textures/misc/00_test_list.txt]: Done for this and the rest.

**Comment by kenrussell** [comment]
Looks good. Thanks Olivia for these tests and the updates.

**Pull request merged**